src/server/game/AuctionHouse/AuctionHouseMgr.cpp

```diff
@@ -752,20 +752,17 @@ bool AuctionHouseObject::BuildListAuctionItems(WorldPacket& data, Player* player
         int loc_idx = player->GetSession()->GetSessionDbLocaleIndex();
         int locdbc_idx = player->GetSession()->GetSessionDbcLocale();
 
-        for (AuctionEntryMap::const_iterator itr = AuctionsMap.begin(); itr != AuctionsMap.end(); ++itr)
+        for (auto &itr: AuctionsMap)
         {
-            if (!AsyncAuctionListingMgr::IsAuctionListingAllowed())                                                    // pussywizard: World::Update is waiting for us...
+            if ((itrcounter++) % 100 == 0) // check condition every 100 iterations
             {
-                if ((itrcounter++) % 100 == 0) // check condition every 100 iterations
+                if (GetMSTimeDiff(GameTime::GetGameTimeMS(), GetTimeMS()) >= searchTimeout) // pussywizard: stop immediately if diff is high or waiting too long
```

Comment:
everytime i see that name i keep thinking that guy must of had some serious confidence.

Reply:
Ah ah indeed